### ISSUE TITLE
Warnings caused by linker flags passed during compilation [unused-command-line-argument]

### ISSUE THREAD

**Issue by robertmaynard** [ROCm-Developer-Tools/HIP#2171]
ROCm 3.7 hip-config file inserts link options on the compile line ( https://github.com/ROCm-Developer-Tools/HIP/blob/rocm-3.7.x/hip-config.cmake.in#L213 )

```cmake
  set_property(TARGET hip::device APPEND PROPERTY
    INTERFACE_COMPILE_OPTIONS -L${HIP_CLANG_INCLUDE_PATH}/../lib/linux -lclang_rt.builtins-x86_64
  )
```

These looks to be causing the resulting compile time warnings 
```
clang-11: warning: -lclang_rt.builtins-x86_64: 'linker' input unused [-Wunused-command-line-argument]
clang-11: warning: argument unused during compilation: '-L/opt/rocm-3.7.0/llvm/lib/clang/11.0.0/../lib/linux/' [-Wunused-command-line-argument]
```

Am I correct in presuming that this application of links flags for compiling is a mistake? If this is needed, I would like to know if we can get this warning removed as it currently breaks `-Werror` builds.

**Comment by yxsamliu** [comment]
that is because that option is passed to device linking command but not used. we will fix that.

**Comment by robertmaynard** [comment]
So to clarify is this link option is required when compiling, or just when doing linking when `gpu-rdc` is enabled?

**Comment by yxsamliu** [comment]
this link option is required when compiling.

**Comment by robertmaynard** [comment]
@yxsamliu Does hipcc also need to be updated ( https://github.com/ROCm-Developer-Tools/HIP/blob/rocm-3.7.x/bin/hipcc#L845 ) as it only adds this when linking?

**Comment by yxsamliu** [comment]
no. this will be fixed in clang driver.

**Comment by yxsamliu** [comment]
sorry, this issue was actually due to hip-config.cmake.in.

Instead of

set_property(TARGET hip::device APPEND PROPERTY
    INTERFACE_COMPILE_OPTIONS -L${HIP_CLANG_INCLUDE_PATH}/../lib/linux -lclang_rt.builtins-x86_64
  )

It should be

set_property(TARGET hip::device APPEND PROPERTY
    INTERFACE_LINK_OPTIONS -L${HIP_CLANG_INCLUDE_PATH}/../lib/linux -lclang_rt.builtins-x86_64
  ) 

we will fix hip-config.cmake.in

**Comment by robertmaynard** [comment]
Closing as this has been fixed in ROCm 4.0
